auction-server/src/auction/service/verification.rs

```diff
@@ -1175,6 +1159,37 @@ impl Service<Svm> {
         Ok(())
     }
 
+    async fn check_memo_instructions(
+        tx: &VersionedTransaction,
+        memo: &Option<String>,
+    ) -> Result<(), RestError> {
+        let memo_instructions = Self::extract_program_instructions(tx, &spl_memo_client::ID);
+        match (memo, memo_instructions.len()) {
+            (None, 0) => Ok(()),
+            (Some(memo), 1) => {
+                let (index, instruction) = memo_instructions[0]; // safe to index because we checked the length
+                if instruction.data != memo.as_bytes() {
+                    return Err(RestError::InvalidInstruction(
+                        Some(index),
+                        InstructionError::InvalidMemoString {
+                            expected: memo.clone(),
+                            found:    String::from_utf8(instruction.data.clone())
+                                .unwrap_or_default(),
+                        },
+                    ));
+                }
+                Ok(())
+            }
+            (Some(_), 0) => Ok(()), // todo: this is for backward compatibility, we should remove this line once searchers have updated their sdk
+            (_, _) => Err(RestError::InvalidInstruction(
```

Comment:
one other thing to think about is what if searchers use memo instructions of their own in the program? there's probably not a reason to fail the tx if it has the required memo program ix and then others on top of that

Reply:
Let's wait until a searcher requests this before doing it.